integrations/weaviate/src/haystack_integrations/document_stores/weaviate/document_store.py

```diff
@@ -124,7 +116,7 @@ def __init__(
         """
         self._client = weaviate.Client(
             url=url,
-            auth_client_secret=auth_client_secret,
+            auth_client_secret=auth_client_secret.resolve_value() if auth_client_secret else None,
```

Comment:
Does it actually make sense to make the credentials optional? Does Weaviate offer any kind of unauthenticated access?

Reply:
Yep, if you check out the tests we do exactly that. Obviously not recommended for production use but it's possible to skip authentication.